doc/branches.md

```diff
@@ -1 +1,1266 @@
+        "fax_number": "5113470",
+        "email": "testingmotor@gmail.com"
+    }
+}
+```
+
+#### Case : Positive Case 3
+
+Request Payload :
+```
+{}
+```
+
+Response HTTP Status Code : 200
+
+Response Payload :
+```
+{
+    "status_code": "CDC-200",
+    "status_message": "No data was changed",
+    "data": {
+        "id": "359ed520-346f-11eb-adc1-0242ac120002",
+        "branch_code": "12345",
+        "company_code": "TES",
+        "branch_name": "Testing Jaya Motor",
+        "address": "Jl. Kebahagiaan 7",
+        "pic": "Budi",
+        "kabeng": "Ridwan",
+        "kelurahan": "Palmerah",
+        "kecamatan": "Palmerah",
+        "kab_kota": "Jakarta Barat",
+        "phone_number": "5367896",
+        "fax_number": "5113470",
+        "email": "testingmotor@gmail.com"
+    }
+}
+```
+
+## <a name="delete-branches"></a>Delete
+
+### Endpoint
+DELETE /branches/{id}
+
+### Headers
+Key | Value 
+--- | ---
+Content-Type | *
+Accept | application/json
+Email | administrator@gmail.com
+
+### Request Payloads
+No Request Payloads
+
+### Response Payloads
+HTTP Code | Status | Description
+--- | --- | --- 
+403 | Forbidden | Branch is not owned by user login
+404 | Not Found | Branch not found in database  
+500 | Internal Server Error | some un-handle error in server 
+204 | No Content | Server request has succeeded but response is no content
+```
+{
+    "status_code": "CDC-403",
+    "status_message": "Branch is not owned by user login",
+    "data": null
+}
+```
+
+Status Code 204 : No Response
+
+### Logic
+
+#### Endpoint validation
+- must be add parameter id
+- id must exist in database
+- must owned by user login
+
+#### Headers Validation
+No Headers Validation
+
+#### Request Payloads Validation
+No Request Payloads Validation
+
+### Scenario Test
+
+#### Case : Negative Case 1
+
+Branch is not owned by user login
+
+Response HTTP Status Code : 403
+
+Response Payload :
+```
+{
+    "status_code": "CDC-403",
+    "status_message": "Branch is not owned by user login",
+    "data": null
+}
+```
+
+#### Case : Negative Case 2
+
+- param id is not exist in database
+
+Endpoint : /branches/359ed520-346f-11eb-adc1-0242ac120112
+
+Response HTTP Status Code : 404
+
+Response Payload :
+```
+{
+    "status_code": "CDC-404",
+    "status_message": "data not found",
+    "data": null
+}
+```
+
+#### Case : Positive Case
+
+- param id is exist in database
+
+Endpoint : /branches/632c725e-35e6-11eb-adc1-0242ac120002
+
+Headers :
+Key | Value 
+--- | ---
+Content-Type | application/json
+Accept | application/json
+Email | administrator@gmail.com
+
+Response HTTP Status Code : 204
+
+Response Payload : No Response
+
+## <a name="get-all-branches"></a>Get All
+
+### Endpoint
+GET /branches
+
+### Headers
+Key | Value 
+--- | ---
+Content-Type | application/json
+Accept | application/json
+Email | administrator@gmail.com
+
+### Request Param
+GET /branches?keyword="testing"&page=1&limit=20&order=branches.id&sort=desc
```

Comment:
all param queries must be in the string. so you don't need to quote marks the value.
GET /branches?keyword=testing&page=1&limit=20&order=branches.id&sort=desc&warehouse_id=1,2,3

Reply:
i think adding company_id filter is better sir

Reply:
o yeah? who will manage the branches menu? user from cdc? or the main dealer user? 
if the one who manages is the user of cdc, I think it is necessary to add a company_id filter. 
But if the administrator is a user from main dealer, branches are automatically filtered based on the company_id of the currently logged-in user.